sdjournal/read.go

```diff
@@ -185,6 +185,10 @@ func (r *JournalReader) Follow(until <-chan time.Time, writer io.Writer) (err er
 	// Process journal entries and events. Entries are flushed until the tail or
 	// timeout is reached, and then we wait for new events or the timeout.
 	var msg = make([]byte, 64*1<<(10))
+	var waitCh = make(chan int)
+	var waitStop = make(chan bool)
```

Comment:
right. No other hidden changes. The main idea of this change is to synchronise sender and receiver to guarantee that only one goroutine is active at a time and use the inner for-loop for waiting for the new entries instead of using the whole `process` loop.